### PR TITLE
Spark integration tests against cloud resources

### PR DESCRIPTION
**Pull request by bijanhoule**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Adds spark integration tests that a user can manually run to test basic spark functionality against local FS and against S3/GCP/Azure cloud buckets.

Out of scope and/or Followup items:

- It'd be nice to add a dimension for testing multiple catalog endpoints as well (or at least parameterize so that different catalog conf could be used for different clouds)
- Shared/ project specific cloud resources (this will require provisioning buckets / secrets management for this project specifically)
    - Github actions automation

### CODE REVIEW

File: integration-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Integration tests
+
+## Prerequisites
+
+- Run `build/sbt clean package publishLocal` to publish spark connector to local maven cache
+- These tests currently assume you have existing cloud resources set up (e.g. S3 bucket & IAM role for S3)
+ 
+## Set up catalog
+
+### Option 1: Use Run localhost unity catalog server
```
Comment (vikrantpuppala) [inline integration-tests/README.md:10]:
```suggestion
### Option 1: Use localhost unity catalog server
```

File: integration-tests/etc/conf/hibernate.properties
Comment (vikrantpuppala) [inline integration-tests/etc/conf/hibernate.properties:1]: do we need this empty file?

Comment (bijanhoule) [reply integration-tests/etc/conf/hibernate.properties:1]: Yeah, IIRC it fails to start if the file doesn't exist.

File: integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkDeltaTableCRUDTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestMethodOrder(OrderAnnotation.class)
+public class SparkDeltaTableCRUDTest extends BaseSparkTest {
+    private static final String BASE_TABLE_NAME = "numbers";
+    private final String RUN_ID = UUID.randomUUID().toString();
+    private static final String SCHEMA = RandomStringUtils.randomAlphabetic(8);
```
Comment (vikrantpuppala) [inline integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkDeltaTableCRUDTest.java:24]: is there a specific reason on why this has to be random?

File: integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkParquetTableCRUDTest.java
Comment (vikrantpuppala) [inline integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkParquetTableCRUDTest.java:1]: also wondering if we should combine the delta and parquet tests into a single file? or is there an advantage to keeping them separate?

**Comment by vikrantpuppala** [comment]
@tdas I was able to run these on my local

File: integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkDeltaTableCRUDTest.java (hunk 2 above)
Comment (bijanhoule) [reply integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkDeltaTableCRUDTest.java:24]: No strong opinion here, it just made it slightly easier for debugging (no conflicts on repeated runs if the server isn't restarted in between runs and failure prevents proper teardown)

File: integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkParquetTableCRUDTest.java
Comment (bijanhoule) [reply integration-tests/src/test/java/io/unitycatalog/integrationtests/SparkParquetTableCRUDTest.java:1]: This was mostly just to keep things simple for the first pass - refactor in the future is totally fine.  There was enough deviation in the behavior of delta and parquet tables that it was easier/cleaner for me to keep them separate for now.

File: build.sbt [hunk 3]
```diff
@@ -551,6 +555,39 @@ lazy val spark = (project in file("connectors/spark"))
...
+      "org.projectlombok" % "lombok" % "1.18.32" % Provided,
+      "org.apache.spark" %% "spark-sql" % "3.5.3" % Test,
+      "io.delta" %% "delta-spark" % "3.2.1" % Test,
+      "org.apache.hadoop" % "hadoop-aws" % "3.3.6" % Test,
+      "org.apache.hadoop" % "hadoop-azure" % "3.3.6" % Test,
+      "com.google.cloud.bigdataoss" % "gcs-connector" % "3.0.2" % Test classifier "shaded",
+      "io.unitycatalog" %% "unitycatalog-spark" % "0.2.0" % Test,
+    ),
+    dependencyOverrides ++= Seq(
+      "com.fasterxml.jackson.core" % "jackson-databind" % "2.15.0",
```
Comment (tdas) [inline build.sbt:579]: @vikrantpuppala @bijanhoule can one of you make a github issue for defining constants for unifying all these common versions?

Comment (bijanhoule) [reply build.sbt:579]: 👍 added here: https://github.com/unitycatalog/unitycatalog/issues/622

**Review by vikrantpuppala — APPROVED** [review]

**Review by tdas — APPROVED** [review]
@bijanhoule this is a fantastic start, but there are a few pieces that are missing. 
- right now, this module in build.sbt is hardcoded to use UC 0.2. but we want that to be easily configurable so that we can do pre-release tests with locally / maven-staged artifacts of arbitrary versions. 
- i would really have liked a single script that the developer can run that does everything, spins up UC, runs the tests with the configuration, and shuts things down. Let's make this a github issue, and hope someone contributes it. Otherwise this will be some pain the next release.
- can we make a github issue to make this into a github workflow?

Despite these gaps, I dont want to block the merging of this PR. We can keep improving this with the help of the community via the github issues.

@bijanhoule thank you for very for building this. this was so crucial in testing out everything in 0.2.

**Pull request merged**